p2pserver/link/link_test.go

```diff
@@ -103,3 +111,113 @@ func TestNewLink(t *testing.T) {
 	}
 
 }
+
+func TestUnpackBufNode(t *testing.T) {
+	cliLink.SetChan(cliChan)
+
+	msgType := "block"
+	var buf []byte
+	var err error
+
+	switch msgType {
+	case "addr":
+		var newaddrs []common.PeerAddr
+		for i := 0; i < 10000000; i++ {
+			newaddrs = append(newaddrs, common.PeerAddr{
+				Time: time.Now().Unix(),
+				ID:   uint64(i),
+			})
+		}
+		var addr mt.Addr
+		addr.NodeAddrs = newaddrs
+		buf, err = addr.Serialization()
+		assert.Nil(t, err)
+	case "consensuspayload":
+		acct := account.NewAccount("SHA256withECDSA")
+		key := acct.PubKey()
+		payload := &mt.ConsensusPayload{
+			Owner: key,
+		}
+		for i := 0; uint32(i) < 200000000; i++ {
+			byteInt := rand.Intn(256)
+			payload.Data = append(payload.Data, byte(byteInt))
+		}
+		buf = payload.ToArray()
+	case "consensus":
+		acct := account.NewAccount("SHA256withECDSA")
+		key := acct.PubKey()
+		payload := &mt.ConsensusPayload{
+			Owner: key,
+		}
+		for i := 0; uint32(i) < 200000000; i++ {
+			byteInt := rand.Intn(256)
+			payload.Data = append(payload.Data, byte(byteInt))
+		}
+		consensus := mt.Consensus{
+			Cons: *payload,
+		}
+		buf, err = consensus.Serialization()
+		assert.Nil(t, err)
+	case "blkheader":
+		var headers []ct.Header
+		blkHeader := &mt.BlkHeader{}
+		for i := 0; uint32(i) < 100000000; i++ {
+			header := ct.Header{}
+			header.Height = uint32(i)
+			header.Bookkeepers = make([]keypair.PublicKey, 0)
+			header.SigData = make([][]byte, 0)
+			headers = append(headers, header)
+		}
+		blkHeader.Cnt = uint32(len(headers))
+		blkHeader.BlkHdr = headers
+		buf, err = blkHeader.Serialization()
+		assert.Nil(t, err)
+	case "tx":
+		var tx ct.Transaction
+		trn := &mt.Trn{}
+		sig := ct.Sig{}
+		sigCnt := 100000000
+		for i := 0; i < sigCnt; i++ {
+			data := [][]byte{
+				{byte(i)},
+			}
+			sig.SigData = append(sig.SigData, data...)
+		}
+		sigs := [1]*ct.Sig{&sig}
+		tx.Payload = new(payload.DeployCode)
+		tx.Sigs = sigs[:]
+		trn.Txn = tx
+		buf, err = trn.Serialization()
+		assert.Nil(t, err)
+	case "block":
+		var blk ct.Block
+		mBlk := &mt.Block{}
+		var txs []*ct.Transaction
+		header := ct.Header{}
+		header.Height = uint32(1)
+		header.Bookkeepers = make([]keypair.PublicKey, 0)
+		header.SigData = make([][]byte, 0)
+		blk.Header = &header
+
+		for i := 0; i < 2400000; i++ {
+			var tx ct.Transaction
+			sig := ct.Sig{}
+			sig.SigData = append(sig.SigData, [][]byte{
+				{byte(1)},
+			}...)
+			sigs := [1]*ct.Sig{&sig}
+			tx.Payload = new(payload.DeployCode)
+			tx.Sigs = sigs[:]
+			txs = append(txs, &tx)
+		}
+
+		blk.Transactions = txs
+		mBlk.Blk = blk
+
+		buf, err = mBlk.Serialization()
+		assert.Nil(t, err)
+	}
+
+	log.Infof("buf size: %d, %dM", len(buf), len(buf)/1024/1024)
+	unpackNodeBuf(cliLink, buf)
```

Comment:
need assert for automated testing. rather than checking the execution log.

Reply:
i mean assert the effects of `unpackNodeBuf`'s execution, clilink.rxBuf.len ==0 , conn is closed ect. or else this test function meaningless.